### PR TITLE
nsync: init at 1.24.0

### PR DESCRIPTION
**Pull request by puffnfresh**
###### Motivation for this change

Library is used in various places, e.g. onnxruntime, Tensorflow

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by puffnfresh** [comment]
@blitz @jonringer PR 1 of 3, final result is here: https://github.com/puffnfresh/nixpkgs/tree/pkgs/obs-backgroundremoval

**Review by blitz — APPROVED** [review]
Looks straightforward. Builds fine. :+1:

**Comment by bobby285271** [comment]
Diff LGTM, fails to build on darwin:

```
mkdir: cannot create directory 'build': File exists
```

Can you look into it (maybe case-insensitivity issue?) or disable the failing platform?

File: pkgs/development/libraries/nsync/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nsync";
+  version = "1.24.0";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "nsync";
```
Comment (sikmir) [inline pkgs/development/libraries/nsync/default.nix:13]: Minor:
```suggestion
    repo = pname;
```

File: pkgs/development/libraries/nsync/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "sha256-jQJtlBDR6efBe1tFOUOZ6awaMTT33qM/GbvbwiWTZxw=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = {
+    homepage = "https://github.com/google/nsync";
+    description = "C library that exports various synchronization primitives";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.puffnfresh ];
```
Comment (sikmir) [inline pkgs/development/libraries/nsync/default.nix:24]:
```suggestion
    maintainers = with lib.maintainers; [ puffnfresh ];
```

**Comment by blitz** [comment]
@tfc This is part of packaging the obs-backgrounremoval plugin. Maybe you can give this a lift?

**Pull request merged**